mp/src/materialsystem/stdshaders/pbr_ps30.fxc

```diff
@@ -304,50 +305,54 @@ float4 main(PS_INPUT i) : COLOR
 
 		ambientLighting = (diffuseIBL + specularIBL) * ambientOcclusion;
 	}
-	// return float4(ambientLighting, 1.0); // testing ambient lighting
 	// End ambient
 
 	// Start direct
 	float3 directLighting = 0.0;
-	for (uint n = 0; n < NUM_LIGHTS; ++n)
-	{
-		float3 LightIn = normalize(PixelShaderGetLightVector(i.worldPos, cLightInfo, n));
-		float3 LightColor = PixelShaderGetLightColor(cLightInfo, n) * GetAttenForLight(i.lightAtten, n); // Li
-
-		directLighting += calculateLight(LightIn, LightColor, outgoingLightDirection,
-				normal, fresnelReflectance, roughness, metalness, lightDirectionAngle, albedo.rgb);
+	if (!FLASHLIGHT) {
+		for (uint n = 0; n < NUM_LIGHTS; ++n)
+		{
+			float3 LightIn = normalize(PixelShaderGetLightVector(i.worldPos, cLightInfo, n));
+			float3 LightColor = PixelShaderGetLightColor(cLightInfo, n) * GetAttenForLight(i.lightAtten, n); // Li
+
+			directLighting += calculateLight(LightIn, LightColor, outgoingLightDirection,
+					normal, fresnelReflectance, roughness, metalness, lightDirectionAngle, albedo.rgb);
+		}
 	}
-	//return float4(directLighting, 1.0); // testing direct lighting
 	// End direct
 
 	// Start flashlight
-    // directLighting = float3(0, 0, 0); // removing direct lighting to test only flashlight
-	if (FLASHLIGHT > 0)
+	if (FLASHLIGHT)
 	{
 		float4 flashlightSpacePosition = mul(float4(i.worldPos, 1.0), g_FlashlightWorldToTexture);
+		clip( flashlightSpacePosition.w );
```

Comment:
this line is to stop projected textures from projecting backwards (only really happens if they have a big FOV because they get frustum culled.)

Reply:
Moved inline